src/sink/rest.rs

```diff
@@ -227,6 +229,11 @@ impl<'de> Deserialize<'de> for SerdeMethod {
 
 #[derive(Clone, Debug, Deserialize)]
 pub struct Config {
+    /// authorization method, if any.
+    /// Options are empty string (default) and "gcp"
+    #[serde(default)]
+    pub auth: String,
```

Comment:
It would be really nice to have a enum here so that we prevent strongly typed data and can validate early :)

Reply:
noted!

Reply:
alright, I'm trying to figure out the best way to handle the enum situation.

I have an enum "Auth" defined like this:
```
/// Authorization methods
#[derive(Serialize, Deserialize)]
enum Auth {
  NoAuth,gcp
}
```
for "gcp" and "NoAuth". I've put them in this order intentionally to avoid false positives from `serde`. more on that in a moment.

My main issue with using this enum is identifying to serde which enum should be used for which situation. I've provided a default for it to fall back to (and put `NoAuth` as the first option)
```

```
    #[serde(default=Auth::NoAuth)]
    pub auth: Auth,
```
HOWEVER -- If I were to deserialize from a `.yaml` file, how would Serde know which `enum Auth` type to deserialize into? If provided with
```
config:
  auth: "gcp"
```
How would it possibly know that I meant the ENUM gcp, and not something else? Looking through [serde's docs on enum representations](https://serde.rs/enum-representations.html) it feels like there might be a disconnect between what serde can discern and what we can expect on our end. At the bottom of the page linked, it states:
> There is no explicit tag identifying which variant the data contains. Serde will try to match the data against each variant in order and the first one that deserializes successfully is the one returned.
for `untagged` variants, which is what we would likely want in the scenario of multiple "constant enums". I guess I'm a little lost on how the deserializer would identify the enum. Happy to talk more or hop on a call sometime to talk it through.

Reply:
@mfelsche  was able to help me out -- got the enum thing worked out with `serde`. Thanks!

Reply:
Weeeh :D nice!